src/coreclr/utilcode/clrhost_nodependencies.cpp

```diff
@@ -355,17 +346,12 @@ void * __cdecl operator new(size_t n, const NoThrow&) NOEXCEPT
     INCONTRACT(_ASSERTE(!ARE_FAULTS_FORBIDDEN()));
 
     void* result = ClrMalloc(n);
-#endif // HAS_ADDRESS_SANITIZER || DACCESS_COMPILE
 	TRASH_LASTERROR;
     return result;
 }
 
-void * __cdecl operator new[](size_t n, const NoThrow&) NOEXCEPT
+void * __cdecl operator new[](size_t n, const std::nothrow_t&) noexcept
```

Comment:
Do we really want the `noexcept` in this case? If we are trying to override the `operator new`, we should match the signature exactly.

Reply:
According to cppreference, the `std::nothrow_t` variants do have `noexcept` in C++11 (they just phrase it really weirdly).

Reply:
Yup, it's also in the shim used by NativeAOT runtime (and portions of GC, JIT that gets compiled): https://github.com/dotnet/runtime/blob/24ded2865fe62786ea4ad4c38f42f6a0083b34e6/src/coreclr/nativeaot/Bootstrap/stdcppshim.cpp#L12

Reply:
Yep. I see that now.